graylog2-web-interface/src/components/cluster-configuration/ClusterConfigurationNodes.test.tsx

```diff
@@ -0,0 +1,142 @@
+    layoutPreferences: {
+      attributes: undefined,
+      order: undefined,
+      pageSize: 0,
+      sort: { attributeId: 'hostname', direction: 'asc' },
+    },
+    searchParams: { sort: { attributeId: 'hostname', direction: 'asc' }, query: '' },
+    isLoadingLayout: false,
+    handleLayoutPreferencesChange: jest.fn(),
+    handleSortChange: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+    (useClusterDataNodesTableLayout as jest.Mock).mockReturnValue(defaultDataLayout);
+    (useClusterGraylogNodesTableLayout as jest.Mock).mockReturnValue(defaultGraylogLayout);
+    (useClusterDataNodes as jest.Mock).mockReturnValue({
+      nodes: [{ id: 'data-1', hostname: 'data-host', metrics: {} }],
+      total: 1,
+      refetch: jest.fn(),
+      isLoading: false,
+      setPollingEnabled: jest.fn(),
+      pollingEnabled: true,
+    });
+    (useClusterGraylogNodes as jest.Mock).mockReturnValue({
+      nodes: [{ id: 'graylog-1', hostname: 'graylog-host', metrics: {} }],
+      total: 1,
+      refetch: jest.fn(),
+      isLoading: false,
+      setPollingEnabled: jest.fn(),
+      pollingEnabled: true,
+    });
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    jest.clearAllMocks();
+  });
+
+  it('renders both node types with default paging and refresh settings in "all" view', () => {
+    render(<ClusterConfigurationNodes />);
+
+    expect(screen.getAllByRole('table')).toHaveLength(2);
+    expect(useClusterDataNodesTableLayout).toHaveBeenCalledWith('', 10);
+    expect(useClusterGraylogNodesTableLayout).toHaveBeenCalledWith('', 10);
+    expect(useClusterDataNodes).toHaveBeenCalledWith(defaultDataLayout.searchParams, { refetchInterval: 5000 });
+    expect(useClusterGraylogNodes).toHaveBeenCalledWith(defaultGraylogLayout.searchParams, { refetchInterval: 5000 });
+  });
+
+  it('switches to a specific node type when segmented control is used', async () => {
+    render(<ClusterConfigurationNodes />);
+
+    fireEvent.click(screen.getByRole('radio', { name: 'Data Nodes' }));
+
+    await waitFor(() => {
+      expect(useClusterDataNodesTableLayout).toHaveBeenLastCalledWith(expect.anything(), 100);
+    });
+
+    expect(useClusterGraylogNodesTableLayout).toHaveBeenCalledTimes(1);
+    expect(useClusterDataNodes).toHaveBeenLastCalledWith(
+      expect.objectContaining({ query: '' }),
+      { refetchInterval: 10000 },
+    );
+  });
+
+  it('uses child "select node type" handler to switch view', async () => {
+    render(<ClusterConfigurationNodes />);
+
+    fireEvent.click(screen.getByRole('radio', { name: 'Data Nodes' }));
+
+    await waitFor(() => {
+      expect(useClusterDataNodesTableLayout).toHaveBeenLastCalledWith('', 100);
+    });
+  });
+
+  it('passes trimmed search query to children', async () => {
+    render(<ClusterConfigurationNodes />);
+
+    const searchInput = screen.getByPlaceholderText('Search nodes…');
+
+    fireEvent.change(searchInput, { target: { value: '  nodes  ' } });
```

Comment:
Here is does not make a big difference, but we usually implement `userEvent` instead of `fireEvent` in new tests.

`fireEvent dispatches DOM events, whereas user-event simulates full interactions, which may fire multiple events and do additional checks along the way.`